src/openrct2/util/Util.cpp

```diff
@@ -204,6 +204,19 @@ static bool cpuid_x86(uint32 * cpuid_outdata, sint32 eax)
 }
 #endif // OPENRCT2_X86
 
+bool sse41_available()
+{
+#ifdef OPENRCT2_X86
```

Comment:
I think this should also check for `__SSE4_1__`. If this is omitted from the build it will try to call SSE 4.1 methods, even though they haven't been compiled in.

Reply:
If this is to check for macro `__SSE4_1__`, it might just as well do `return true;` and throw runtime detection out the window

Reply:
What on earth are you on about? What use is runtime detection if the necessary code is not compiled in? It should just return false if `__SSE4_1__` is not defined.

Reply:
The only way this could have `__SSE4_1__` defined, is if you enabled SSE4.1. If you enable it for `Util.cpp`, you might just as well enable it for the whole project and thus require SSE4.1 and do away with runtime detection, because other parts of your code could rely on it already.

So you either do runtime detection as I have implemented it or enable SSE4.1 project-wise and so `return true;` here.

Reply:
I think what @Gymnasiast is wants is a check for if the SEE4.1 function was compiled in or not and if it isn't, don't do a runtime check. So if you would be something like:

```
#ifdef __SEE4_1_ENABLED__
    ...
#endif
```

Reply:
I don't see a point in any of this, we either do runtime detection or not. If you want, you can do `#if defined(OPENRCT2_X86) #define __SSE4_1_ENABLED__ #endif`, but I don't see what other benefits there are but saving a single call to `cpuid`.

Reply:
It could serve its purpose if we checked for `__SSE4_1__` macro, for users that set their cflags and cxxflags to e.g. `-march=native`, but MSVC breaks that, the way it is done now.

Reply:
The reason I mention this is the following: if you forget to pass the SSE 4.1 flag to the compiler, like currently happens on macOS, a binary will compiled that will check for SSE 4.1 capability, and then crash because the SSE 4.1 drawing code is not compiled in.